### PR TITLE
Bloq Quickstarter

### PR DESCRIPTION
**Pull request by mpharrigan**
A single-page vanilla-javascript little app that will produce some of the boilerplate for bootstrapping a new bloq.

image

### CODE REVIEW

**Review by fdmalone — CHANGES_REQUESTED** [review]
very cool

File: dev_tools/bloq-quickstarter.html [hunk 1]
```diff
@@ -0,0 +1,274 @@
...
+        ${signature}
+
+    def build_composite_bloq(self, bb: 'BloqBuilder'${build_sig}) -> Dict[str, 'SoquetT']:
+        raise NotImplementedError("Implement or delete.")
+
+    def bloq_counts(self, ssa: Optional['SympySymbolAllocator'] = None) -> Set['BloqCountT']:
+        raise NotImplementedError("Implement or delete.")
+
+    def short_name(self) -> str:
+        return f'${className}'
```
[the remark this answers is not in the document]
Comment (mpharrigan) [reply dev_tools/bloq-quickstarter.html:249]: is that a thing that happens?

[the remark this answers is not in the document]
Comment (fdmalone) [reply dev_tools/bloq-quickstarter.html:249]: (copied code to gen_bloq.py and ran format_incremental)
gen_bloq.py:34:15: W1309: Using an f-string that does not have any interpolated variables (f-string-without-interpolation)
gen_bloq.py:2:0: W0611: Unused Union imported from typing (unused-import)
gen_bloq.py:6:0: W0611: Unused Side imported from qualtran (unused-import)

Anyway, this is not really important, it's mean to be a quickstart, not a complete bloq.

[the remark this answers is not in the document]
Comment (mpharrigan) [reply dev_tools/bloq-quickstarter.html:249]: I'll remove the fstring part. The unused imports are sortof part of it as it's easier to prune unused imports than add ones that are missing -- at least in the context of a quickstarter

**Review by fdmalone — APPROVED** [review]
LGTM

**Comment by fdmalone** [comment]
Are these bot warnings something to worry about?

File: dev_tools/bloq-quickstarter.html [hunk 2]
```diff
@@ -0,0 +1,274 @@
...
+`;
+
+      document.getElementById("template_code").innerHTML = template;
+      hljs.highlightAll();
+    }
+  </script>
+
+  <!-- Syntax highlighting -->
+  <link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/highlight.js/11.8.0/styles/default.min.css">
+  <script src="https://cdnjs.cloudflare.com/ajax/libs/highlight.js/11.8.0/highlight.min.js"></script>
```
[the remark this answers is not in the document]
Comment (mpharrigan) [reply dev_tools/bloq-quickstarter.html]: fixed

**Comment by mpharrigan** [comment]
I'm pretty sure the fact that I'm putting all the user input through `hljs.highlight` and its wrapped in `<pre><code>` means you can't inject anything nefarious. In fact-- I put in a sanitizer but then whenever you would type a "<" in the input field it would literally turn up as `&lt;` in the code block.

I changed the way the code is organized so maybe the analyzer can see that all user input is being funneled through `hljs.highlight` which hopefully it knows does sanitization.

**Comment by mpharrigan** [comment]
the bot is happy yay i'm basically a frontend dev now

**Pull request merged**